arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,368 @@
+---
+title: "Quickstart: Build a decentralized app with Arbitrum (Solidity, Hardhat)"
+sidebar_label: Quickstart (Solidity, Hardhat)
+description: This quickstart will prepare you to build your first decentralized app (dApp / web3 app) by speedwalking you through the process of deploying a simple Solidity smart contract directly to Arbitrum.
+author: symbolpunk
+author-objective: build a shareable quickstart that onboards web2 developers into Arbitrum
+reader-audience: web developers who haven't ever built on Ethereum/Arbitrum
+reader-task: deploy my first smart contract directly to L2 with a confident understanding of Arbitrum's value prop
+content-type: quickstart
+todos:
+  1: footnotes
+  2: faq
+  3: quicklooks
+  4: attach to behavior
+  5: demonstrate "funnel testing"
+  6: demonstrate other patterns / components / best practices
+  7: meet partnerships and marketing needs
+  8: codepen?
+  9: samples repo?
+  10: public preview
+  11: default Build DApps to this page
+---
+
+import PublicPreviewBannerPartial from '../partials/_public-preview-banner-partial.md'; 
+
+<PublicPreviewBannerPartial />
+
+This quickstart is for web developers who want to start building <a data-quicklook-from='decentralized-application'>decentralized applications</a> (dApps) using <a data-quicklook-from='arbitrum'>Arbitrum</a>. It makes no assumptions about your prior experience with Ethereum, Arbitrum, or Solidity. Familiarity with Javascript and package managers like `npm` and/or `yarn` is expected. If you're new to Ethereum, consider studying the [Ethereum documentation](https://ethereum.org/en/developers/docs/) before proceeding.
+
+### What we're building
+
+We're going to build a digital cupcake vending machine using Solidity smart contracts[^1]. Our vending machine will follow two rules:
+
+1. The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+2. The vending machine's rules can't be changed by anyone.
+
+Here's our vending machine implemented with Javascript:
+
+import { VendingMachine } from '@site/src/components/VendingMachine/VendingMachine'
+
+<VendingMachine id='dumb-cupcakes' type='web2' />
+
+Note that although this vending machine follows Rule 1, it doesn't follow Rule 2 (yet)[^2].
```

Comment:
Should we explicitly mention why it doesn't follow rule 2? (It is described below, but at this point it might make the user feel a bit lost because it might not understand how/why the rules can be changed).